vmdb/app/models/job.rb

```diff
@@ -173,16 +173,19 @@ def timeout!
   def self.check_jobs_for_timeout
     $log.debug "Checking for timed out jobs"
     begin
-      self.in_my_region.find(:all, :conditions => ["((state != 'finished' and state != 'waiting_to_start') or (state = 'waiting_to_start' and dispatch_status = 'active')) and (zone is null or zone = ?)", MiqServer.my_zone]).each do |job|
-        if job.updated_on < current_job_timeout.seconds.ago
+      in_my_region
+        .where("state != 'finished' and (state != 'waiting_to_start' or dispatch_status = 'active')")
+        .where("zone is null or zone = ?", MiqServer.my_zone)
```

Comment:
Yeah, I got a bit slack on working out the actually-best spelling for some of these